pkg/operator/controller/nodeobservability/controller.go

```diff
@@ -68,16 +64,13 @@ type NodeObservabilityReconciler struct {
 //+kubebuilder:rbac:urls=/node-observability-pprof,verbs=get;
 //+kubebuilder:rbac:groups=authentication.k8s.io,resources=tokenreviews,verbs=create;
 //+kubebuilder:rbac:groups=authorization.k8s.io,resources=subjectaccessreviews,verbs=create;
-//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles,verbs=list;get;create;watch;delete;
-//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterrolebindings,verbs=list;get;create;watch;delete;
-//+kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,verbs=list;get;create;watch;use;delete;
-//+kubebuilder:rbac:groups=apps,namespace=node-observability-operator,resources=daemonsets,verbs=list;get;create;watch;update;patch;
-//+kubebuilder:rbac:groups=core,namespace=node-observability-operator,resources=secrets,verbs=list;get;create;watch;delete;
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles,verbs=get,resourceNames=node-observability-operator-controller-role
```

Comment:
I believe you need to add `bind` verb too, to restrict the possibility to bind only to the resource `node-observability-operator-controller-role`.

Reply:
Just checked this, actually `bind` verb is only needed when the privileged escalation needs to be bypassed. If your code is working (manged to create the clusterrolebinding for the agent role) I think it's not needed then.